lib/fuse_lowlevel.c

```diff
@@ -320,38 +320,46 @@ int fuse_reply_iov(fuse_req_t req, const struct iovec *iov, int count)
 	return res;
 }
 
-static size_t fuse_dirent_size(size_t namelen)
-{
-	return FUSE_DIRENT_ALIGN(FUSE_NAME_OFFSET + namelen);
-}
+#ifndef IFTODT
+# define IFTODT(mode) (((mode) & 0170000) >> 12)
+#endif
```

Comment:
IFTODT is a standard macro as defined in dirent.h.

http://www.gnu.org/software/libc/manual/html_node/Directory-Entries.html